### ISSUE TITLE
/managed/resources/export exports incorrect deployWith strategy

### ISSUE THREAD

**Issue by coryb** [spinnaker/keel#753]
My cluster is currently managed with:
```
     deployWith:
       strategy: highlander
```
but if I export the same cluster it returns with the default redblack:
```
      deployWith:
        delayBeforeDisable: PT0S
        delayBeforeScaleDown: PT0S
        maxServerGroups: 2
        resizePreviousToZero: false
        rollbackOnFailure: true
        strategy: red-black
```

The export should honor the currently managed deployWith strategy.

Ideally we would be able to inspect current deployment pipelines for clusters to infer the correct strategy when exporting a non-managed cluster to be managed, but for now we need it to be consistent with the currently managed strategy.

**Comment by spinnakerbot** [comment]
This issue hasn't been updated in 58 days, so we are tagging it as 'stale'. If you want to remove this label, comment:

> @spinnakerbot remove-label stale

**Comment by luispollo** [comment]
@coryb Export was originally thought was a feature you would only ever use with _non_-managed resources, and, since we cannot infer the deployment strategy from clusters, only pipelines, we defaulted (hard-coded, really) to red-black on export.

We could change this behavior as per your suggestions above, but first, do you mind giving us a bit more context on what use case(s) you're trying to support with an export of an already-managed resource?

**Comment by coryb** [comment]
hey @luispollo.  On the initial export where a user wants to go from non-managed to managed, it will definitely catch people by surprise to export a cluster and have the deployment strategies change from what they are currently using.  It will further confuse users if they try to run the `diff` and it shows no changes (despite the strategy changing).  I thought @robfletcher had said there was a way to infer the deployment strategy from the pipeline or task used to deploy the current cluster.  

Consistency for managed clusters is only really necessary if I re-export the cluster. I had been doing that a bunch as your apis change through the alpha period, and I thought it was useful.  Admittedly this use case will diminish as the api become stable.  It just seems like good hygiene that if a cluster is managed and I re-export it that you export the currently used strategy, you clearly already know what the strategy since it is tracked in the previously posted delivery config.

**Comment by luispollo** [comment]
Thanks @coryb, that's very helpful! We can indeed figure out the deployment strategy from pipelines. In fact, we just spent some time earlier today discussing an experiment I did that looked solely at pipelines (as opposed to resources) to export a full delivery config (more in #1007 if you're curious, but we've decided not to pursue this for now). It will be interesting to see if some of the code I wrote for that can be reused to address this issue -- I think so!
